### ISSUE TITLE
Feature request: about 5.x transformer.

### ISSUE THREAD

**Issue by wanhmr** [SDWebImage/SDWebImage#2542]
### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / macos / tvos / watchos
 Platform Version        | e.g. 11.0 / 10.13.0 / 11.0 / 4.0
 SDWebImage Version      | e.g. 4.2.0 / 4.1.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 9 / Xcode 8
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

### Description

I noticed that 5.x provides the feature a transformer. This is a great feature, I like it very much. But I think it can be even better.

Now this feature will only save the transformed image, but the original image may be used in multiple places and with different transformers applied.

So I hope to a option the original image can be saved at the same time.

Best wishes.

**Comment by dreampiggy** [comment]
> Now this feature will only save the transformed image, but the original image may be used in multiple places and with different transformers applied.

This is a Good idea. I can make it as a feature request.

But this need some more careful though about this. Once we save both of two image for cache, we need to provide more about our current cache key system because now, even if the transformed cache key miss, we need query original image and do transform JUST IN TIME. Which may block the caller queue (such as memory cache query).

I'll check back later this week. It seems that I'm busy this days. But maybe it's time to go back this project. 🍺

**Comment by dreampiggy** [comment]
I write a demo solution. Please have a check at: https://github.com/dreampiggy/SDWebImage/commit/0f00067aa4d4985a1bf7b6318e190360d6894e55

This allows you to decide where to store the original image, as well as the transformed image. Like:

```swift
let context = [.storeCacheType : .all, .originalStoreCacheType : .disk] // transformed image to both memory && disk, original image to disk only
sd_setImage(with:url options:[] context:context])
```

**Comment by wanhmr** [comment]
@dreampiggy Thx. I'll use it later.

**Comment by dreampiggy** [comment]
@stale Fresh
